### PR TITLE
feat: add dynamic section

### PR DESCRIPTION
**Pull request by Meierschlumpf**
<br/>
<div align="center">
  
  <h3>Homarr</h3>
</div>

**Thank you for your contribution. Please ensure that your pull request meets the following pull request:**

- [x] Builds without warnings or errors (``pnpm buid``, autofix with ``pnpm format:fix``)
- [x] Pull request targets ``dev`` branch
- [x] Commits follow the [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] No shorthand variable names are used (eg. ``x``, ``y``, ``i`` or any abbrevation)

Current issues I'm facing:

- [x] minRow would be a nice addition so that it would always have the full grid available to put items inside and not just the first row (or if items are there the rows where items are). I tried it with setting the minRow property in options again when the height of the section changed. But until now without success
- [x] Calculation of size is not quite good yet:

image

- [x] Also while dragging the items are a little big:

image

- [x] And this one should be improved:

image

image
image

### CODE REVIEW

File: apps/nextjs/src/components/board/sections/content.tsx [hunk 1]
```diff
@@ -1,227 +1,47 @@
...
+  }, [section.items, innerSections]);
 
   return (
     <>
-      {items.map((item) => (
-        <BoardItem key={item.id} refs={refs} item={item} opacity={board.opacity} />
+      {sortedItems.map((item) => (
+        <GridStackItem
+          key={item.id}
+          innerRef={refs.items.current[item.id] as React.RefObject<HTMLDivElement>}
```
Comment (Meierschlumpf) [inline apps/nextjs/src/components/board/sections/content.tsx:31]: May be undefined

File: apps/nextjs/src/components/board/sections/content.tsx [hunk 2]
```diff
@@ -1,227 +1,47 @@
...
+          width={item.width}
+          height={item.height}
+          xOffset={item.xOffset}
+          yOffset={item.yOffset}
+          kind={item.kind}
+          id={item.id}
+          type={item.type}
+          minWidth={item.type === 'section' ? Math.max(...item.items.map(item => item.xOffset + item.width), ...board.sections.filter((section): section is DynamicSection => section.kind === 'dynamic' && section.parentSectionId === item.id).map(item => item.xOffset + item.width), 1) : undefined}
```
Comment (Meierschlumpf) [inline apps/nextjs/src/components/board/sections/content.tsx:39]: Improve

File: apps/nextjs/src/components/board/sections/section-actions.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          };
+        }),
+      }));
+    },
+    [updateBoard],
+  );
+
+  const moveInnerSectionToSection = useCallback(
+    ({ innerSectionId, sectionId, ...positionProps }: MoveInnerSectionToSection) => {
+      console.log("moveInnerSectionToSection", innerSectionId, sectionId, positionProps);
```
Comment (Meierschlumpf) [inline apps/nextjs/src/components/board/sections/section-actions.tsx:44]: Remove `console.log`

**Comment by Meierschlumpf** [comment]
Calculation of size solved:

image

File: apps/nextjs/src/components/board/sections/gridstack/init-gridstack.ts [hunk 4]
```diff
@@ -15,20 +16,21 @@ interface InitializeGridstackProps {
...
-export const initializeGridstack = ({ section, refs, sectionColumnCount }: InitializeGridstackProps) => {
+export const initializeGridstack = ({ section, itemIds, refs, sectionColumnCount }: InitializeGridstackProps) => {
   if (!refs.wrapper.current) return false;
   // initialize gridstack
   const newGrid = refs.gridstack;
   newGrid.current = GridStack.init(
     {
       column: sectionColumnCount,
-      margin: Math.round(Math.max(Math.min(refs.wrapper.current.offsetWidth / 100, 10), 1)),
+      margin: 10, // Math.round(Math.max(Math.min(refs.wrapper.current.offsetWidth / 100, 10), 1)),
```
Comment (manuel-rw) [inline apps/nextjs/src/components/board/sections/gridstack/init-gridstack.ts:26]: Remove commented out code

File: apps/nextjs/src/components/board/sections/gridstack/use-gridstack.ts [hunk 5]
```diff
@@ -53,44 +84,89 @@ export const useGridstack = ({ section, mainRef }: UseGridstackProps): UseGrista
...
+          itemId: id,
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          xOffset: changedNode.x!,
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          yOffset: changedNode.y!,
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          width: changedNode.w!,
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          height: changedNode.h!,
+        });
```
Comment (manuel-rw) [inline apps/nextjs/src/components/board/sections/gridstack/use-gridstack.ts:104]: Explain with a comment why you suppress the error

**Review by manuel-rw — CHANGES_REQUESTED** [review]
Looks good. Would be nice if you add more comments - the Gridstack API is very complex and I think it would be useful if the "bar for entry" would be lower - even I can't follow all the time.

Also, would you be available to quickly test & try this out together in a call before we merge this? Thank you for your work 🚀❤️

**Review by manuel-rw — APPROVED** [review]

**Comment by Meierschlumpf** [comment]
> Also, would you be available to quickly test & try this out together in a call before we merge this? Thank you for your work 🚀❤️

sure

**Review by manuel-rw — APPROVED** [review]

**Pull request merged**